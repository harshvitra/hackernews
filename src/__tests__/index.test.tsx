import { render, screen } from "@testing-library/react";
import App, { getStaticProps } from "../../pages/index";
import { mount } from "enzyme";
const stories = [100, 200, 300]
const wrapper = mount(<App stories={stories} />);

describe("Testing Home page", () => {

  it("renders without crashing", () => {
    render(<App stories={stories} />);
    expect(
      screen.getByRole("heading", { name: "Hacker News" })
    ).toBeInTheDocument();
  });

  it("It should render the component to match the snapshopt", () => {
    expect(wrapper).toMatchSnapshot()
  });

  it("Scroll behaviour of the list", () => {
    const mEvent = {
      target: { clientHeight: 650, scrollHeight: 1200, scrollTop: 400 },
    };
    wrapper.find('#storiesList').simulate('scroll', mEvent);
  });

  // Could not make this one to work - css modules are mocked
  // it("should click on the theme button and simulate click", () => {
  //   wrapper.find("#themeButton").simulate('click');
  //   expect(wrapper.find("#container").props().className).toContain("darkMode")
  // });

  // Attemp to try API testing - failed 
  // it('should load user data', () => {
  //   return getStaticProps()
  //     .then(stories => {
  //       expect(stories).toBeDefined()
  //     })
  // })
});
