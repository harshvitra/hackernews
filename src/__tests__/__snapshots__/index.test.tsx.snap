// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home page It should render the component to match the snapshopt 1`] = `ReactWrapper {}`;
